packages/vm/src/runBlock.ts

```diff
@@ -918,3 +950,24 @@ const DAOConfig = {
   ],
   DAORefundContract: 'bf4ed7b27f1d666546e30d74d50d173d20bca754',
 }
+
+/**
+ * This helper method generates a list of all CL requests that can be included in a pending block
+ * @param _vm VM instance from which to derive CL requests
+ * @returns an list of CL requests in ascending order by type
+ */
+export const accumulateRequests = async (_vm: VM): Promise<CLRequest[]> => {
+  const requests: CLRequest[] = []
+
+  // TODO: Add in code to accumulate deposits (EIP-6110)
+
+  // TODO: Add in code to accumulate partial withdrawals (EIP-7002)
```

Comment:
Simply for my understand: so here a call into `read_withdrawal_requests()` from https://eips.ethereum.org/EIPS/eip-7002#execution-layer ("System Call" section) would take place, right?

Side question for @acolytec3, since you likely have better imagination on this: how does this affect execution with e.g. RPCStateManager? Does this still work (do not yet get the pieces together in my head)?

Reply:
To the question for me, I don't "think" this should have any impact.  We already track contract state updates in a local cache in the RPCStateManager so it should keep track of any changes we make in the processing of requests.

Reply:
@holgerd77 Yes exactly here the call to `read_withdrawal_requests` is made. For 6110 I am not sure if all the logic is in here or not (but I am not super deep into 6110, @scorbajio might have better insights here)

Reply:
For 6110, deposits are retrieved from the block receipt logs for any logs whose address is `DEPOSIT_CONTRACT_ADDRESS` and also from the block body requests field for any requests that are typed as deposit requests.